server/service/src/item_stats.rs

```diff
@@ -82,6 +82,54 @@ pub fn get_item_stats(
     ))
 }
 
+pub fn get_item_stats_map(
+    connection: &StorageConnection,
+    store_id: &str,
+    amc_lookback_months: Option<f64>,
+    item_ids: Vec<String>,
+) -> Result<HashMap<String, ItemStats>, PluginOrRepositoryError> {
+    let default_amc_lookback_months =
+        get_store_preferences(connection, store_id)?.monthly_consumption_look_back_period;
+
+    let amc_lookback_months = match amc_lookback_months {
+        Some(months) => months,
+        None => default_amc_lookback_months,
+    };
+
+    let consumption_map = get_consumption_map(
+        connection,
+        store_id,
+        item_ids.clone(),
+        amc_lookback_months,
+    )?;
+
+    let input = amc::Input {
+        store_id: store_id.to_string(),
+        amc_lookback_months,
+        // Really don't like cloning this
+        consumption_map: consumption_map.clone(),
+        item_ids: item_ids.clone(),
+    };
+
+    let amc_by_item = match PluginInstance::get_one(PluginType::AverageMonthlyConsumption) {
+        Some(plugin) => amc::Trait::call(&(*plugin), input),
+        None => amc::Trait::call(&DefaultAmc, input),
+    }?;
+
+    let item_stats_vec = ItemStats::new_vec(
+        amc_by_item,
+        consumption_map,
+        get_stock_on_hand_rows(connection, store_id, Some(item_ids))?,
+    );
+
+    let item_stats_map = item_stats_vec
```

Comment:
Everything above this line looks identical to `get_item_stats`... could could just do:

```rs
let item_stats_vec = get_item_stats(bla);
let item_stats_map = ...
```